src/group/mod.rs

```diff
@@ -109,6 +109,7 @@ impl Group {
 
     /// Creates the Klein 4 group
     pub fn klein_4() -> Self {
+        //TODO fix this as this isn't K_4
```

Comment:
The Klein Group should be isomorphic to the dihedral group of order 4, if it isn't it's probably a bug in D_2n?

Reply:
Yes I think it is a bug with the generation method for D_2n not working for n < 3 (I believe). I didn't fix that in this, but was going to open a pull request for that.